arch/posix/core/cpuhalt.c

```diff
@@ -45,17 +43,3 @@ void arch_cpu_atomic_idle(unsigned int key)
 	sys_trace_idle();
 	posix_atomic_halt_cpu(key);
 }
-
-#if defined(CONFIG_REBOOT)
-/**
- * @brief Stub for sys_arch_reboot
- *
- * Does nothing
- */
-void __weak sys_arch_reboot(int type)
```

Comment:
I may be missing something, but instead of removing this, shouldnt be select HAS_REBOOT from ARCH_POSIX, and keep this (now named `z_sys_reboot()`)

Reply:
I guess it's ok to allow posix implementation to shut the application down because it cannot reboot, then? A reboot in posix would be like `sys_reboot()` + relaunch zephyr manually?

Reply:
Yep, this default reboot hook is just terminating returning 1 to the calling program after printing a descriptive message. A board, or app developer can override that with something else.
Normally a reboot is the result of a failure. And in an environment like this, terminating by default is reasonable.

Reply:
please check latest revision. I've removed __weak (non-standard feature_ in favor of a Kconfig option (like ARM)